PyNvCodec/TC/src/Tasks.cpp

```diff
@@ -513,37 +513,26 @@ int NvdecDecodeFrame::GetCapability(NV_DEC_CAPS cap) const
 
 namespace VPF
 {
-auto const format_name = [](Pixel_Format format) {
-  stringstream ss;
+#define TC_PIXEL_FORMAT_STRINGIFY(s) TC_PIXEL_FORMAT_STRINGIFY_(s)
+#define TC_PIXEL_FORMAT_STRINGIFY_(s) #s##
+#define TC_PIXEL_FORMAT_ENUM_CASE(s)                                           \
+  case s:                                                                      \
+    return TC_PIXEL_FORMAT_STRINGIFY(s)
 
+auto const format_name = [](Pixel_Format format) {
   switch (format) {
-  case UNDEFINED:
-    return "UNDEFINED";
-  case Y:
-    return "Y";
-  case RGB:
-    return "RGB";
-  case NV12:
-    return "NV12";
-  case YUV420:
-    return "YUV420";
-  case RGB_PLANAR:
-    return "RGB_PLANAR";
-  case BGR:
-    return "BGR";
-  case YCBCR:
-    return "YCBCR";
-  case YUV444:
-    return "YUV444";
-  case RGB_32F:
-    return "RGB_32F";
-  case RGB_32F_PLANAR:
-    return "RGB_32F_PLANAR";
-  case YUV422:
-    return "YUV422";
+    TC_PIXEL_FORMAT_ENUM_CASE(UNDEFINED);
+    TC_PIXEL_FORMAT_ENUM_CASE(Y);
+    TC_PIXEL_FORMAT_ENUM_CASE(RGB);
+    TC_PIXEL_FORMAT_ENUM_CASE(NV12);
+    TC_PIXEL_FORMAT_ENUM_CASE(YUV420);
+    TC_PIXEL_FORMAT_ENUM_CASE(YCBCR);
+    TC_PIXEL_FORMAT_ENUM_CASE(YUV444);
+    TC_PIXEL_FORMAT_ENUM_CASE(RGB_32F);
+    TC_PIXEL_FORMAT_ENUM_CASE(RGB_32F_PLANAR);
+    TC_PIXEL_FORMAT_ENUM_CASE(YUV422);
   default:
-    ss << format;
-    return ss.str().c_str();
+    return "Unknown Pixelformat";
```

Comment:
could be a form of `__builtin_unreachable`.

Reply:
Before we got a more informative string back containing the `format` that was teh input argument right ? I would say this makes it easier to find the error than a less informative Unknown pixel format.

Reply:
we could return a `std::string` or throw or exit. In any case, it would be a bug (either in the user code constructing an invalid variant or us forgetting a variant here). `PixelFormat` can only have these values. Returning a number contains no information it just hides a bug. With deleted default case we would get a warning

Go for throwing or aborting when we hit default?

Reply:
Ideally, PixelFormat would be a `enum class`.

Reply:
Oh right nevermind i am onboard with throwing. 👍